### PR TITLE
FP-0013 pagination

### PR DESCRIPTION
**Pull request by TesaRocks**
Hi Diego, please have a look at my PR
Once you agree, I can implement the same logic to UserList component.

I did it mostly by myself since I could not find many suitable answers on the web.

Regards

Alexis

### CODE REVIEW

File: src/app/invoice/invoice.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<mat-grid-list cols="2" rowHeight="450px">
+  <mat-grid-tile>
+    <mat-card class="product-card">
```
Comment (diegolaciar) [inline src/app/invoice/invoice.component.html]: I dont get what is this component.

File: src/app/aplication-state.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
+import { IInvoiceDetailInitialState } from './invoice/invoice-detail/ngrx/invoice-detail.reducer';
+import { IInvoiceInitialState } from './invoice/invoice-list/ngrx/invoice-list.reducer';
 import { IProductsInitialState } from './products/ngrx/product.reducer';
 import { IUsersInitialState } from './users/ngrx/user.reducer';
 
 export interface IApplicationState {
   userState: IUsersInitialState;
   productState: IProductsInitialState;
+  invoiceState: IInvoiceInitialState;
+  invoiceDetailState: IInvoiceDetailInitialState;
```
Comment (diegolaciar) [inline src/app/aplication-state.ts]: I don't like this idea, this is not a good design
You should handle all the invoices and the detail in ONE model.

**Review by diegolaciar — CHANGES_REQUESTED** [review]
hi @TesaRocks,

This PR is huge! 
And it will be very affected by the change of the invoice model.
Please work on that, then send a new PR.

Cheers,
Diego

File: src/app/aplication-state.ts (hunk 2 above)
Comment (TesaRocks) [reply src/app/aplication-state.ts]: done thnaks

File: src/app/invoice/invoice.component.html (hunk 1 above)
Comment (TesaRocks) [reply src/app/invoice/invoice.component.html]: fixed thanks

**Pull request merged**